build.sh

```diff
@@ -10,9 +10,17 @@
 
 set -ex
 root="$(cd $(dirname ${BASH_SOURCE[0]}) && pwd)"
-if [[ -z ${CC} ]]; then export CC=/usr/bin/gcc; fi
-if [[ -z ${CXX} ]]; then export CXX=/usr/bin/g++; fi
-if [[ ! -z ${CCACHE_DIR} && -e /dev/shm ]]; then export CCACHE_DIR=/dev/shm/redpanda; fi
+if [[ -z ${CC} ]]; then export CC=clang; fi
+if [[ -z ${CXX} ]]; then export CXX=clang++; fi
+if [[ -z ${DEPOT_TOOLS_DIR} ]]; then DEPOT_TOOLS_DIR=/opt/depot_tools; fi
+if [[ -z ${CCACHE_DIR} && -e /dev/shm ]]; then
+  mkdir -p /dev/shm/redpanda
+  export CCACHE_DIR=/dev/shm/redpanda
+fi
+
+ccache -p # print the config
+ccache -s # print the stats before reusing
```

Comment:
```
2021-11-01T01:22:14.2219376Z max cache size                     200.0 MB
```
Is this sufficient?

Reply:
i don't know tbh, i haven't profiled after the changes in this pr

Reply:
Hmm, I'm not convinced it's working:
```
files in cache                       732
cache size                          29.7 MB
max cache size                       5.0 GB
```